src/pal/src/misc/perftrace.cpp

```diff
@@ -659,26 +659,29 @@ PERFlushAllLogs( )
 
 static
 void
-PERFLogFileName(PathCharString * destFileString, const char *fileName, const char *suffix, int max_length)
+PERFLogFileName(PathCharString& destFileString, const char *fileName, const char *suffix)
 {
     const char *dir_path;
     CPalThread* pThread = InternalGetCurrentThread();
-    char * destFileName = (char*)InternalMalloc(pThread, max_length);
     dir_path = (profile_log_path == NULL) ? "." : profile_log_path;
-
+        
+    destFileString.Append(dir_path, strlen(dir_path));
+    destFileString.Append(PATH_SEPARATOR, strlen(PATH_SEPARATOR));
     if (fileName != NULL)
     {
-        snprintf(destFileName, max_length, "%s%s%s", dir_path, PATH_SEPARATOR, fileName);
+        destFileString.Append(fileName, strlen(fileName));
     }
     else
     {
-        snprintf(destFileName, max_length, "%s%s%d_%d%s", dir_path, PATH_SEPARATOR,
-            program_info.process_id, THREADSilentGetCurrentThreadId(), suffix);
+        std::string process_id     = std::to_string(program_info.process_id);
```

Comment:
Ah, I am sorry for missing the fact it converts int to string. But even if it is not being compiled, I don't think putting stl stuff here is ok. Someone can see that and assume it is really ok to use stl in the PAL.
Looking at the code again, it seems that keeping the original snprintf, estimating the needed string length first and using it as a parameter to OpenStringBuffer would be better for long paths since it would not require repeated reallocations of the buffer in the Append. For the estimation, you can use the max possible length of a printed integer.